src/qmxgraph/page/utils.js

```diff
@@ -66,16 +66,57 @@ graphs.utils.setStyleKey = function setStyleKey (style, key, value) {
         }
         else
         {
-            var cont = style.indexOf(';', index);
+            var parts = style.split(';')
+            for (var i = 0; i < parts.length; ++i)
+            {
+                var current = parts[i];
+                var current_key = current.split('=')[0];
+                if (current_key == key)
+                {
+                    // Just to find the key index
+                    break;
+                }
+            }
 
-            style = style.substring(0, index) + key + '=' + value +
-                ((cont >= 0) ? style.substring(cont) : '');
```

Comment:
Bugfix. This was matching wrong keys in some scenarios. Example:
```
>>> graphs.utils.setStyleKey('foobar=3;bar=4', 'bar', 7)
"foobar=7;bar=4"
```
The expected is to change `bar` key, but it matches the end of the first key and updates that one instead.

Reply:
Good catch!

Reply:
This was actually @prusse-martin who found, kudos to him.